mp/src/game/server/momentum/mom_triggers.cpp

```diff
@@ -1570,6 +1570,9 @@ void CNoGrenadesZone::OnStartTouch(CBaseEntity* pOther)
         return;
 
     pSticky->SetCanExplode(false);
+
+    if (m_iStickyFizzleType == FIZZLE_ON_ENTRANCE)
+        pSticky->Destroy(true);
```

Comment:
The main thing about what I'm requesting is this means this can apply to rockets as well. So you'll want to instead cast to `CMomExplosive` instead, it should still have a destroy method to use here.

Reply:
Making the `pSticky` variable to instead cast to `CMomExplosive` means I can't use `SetCanExplode()`, should I make a new variable?

Reply:
Oh right yep you'll want to keep the cast just for that.